kms/key/kms.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"github.com/acorn-io/aws/kms/key/props"
+	"github.com/acorn-io/services/aws/libs/common"
+	"github.com/aws/aws-cdk-go/awscdk/v2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsiam"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awskms"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/sirupsen/logrus"
+)
+
+func newKMSKeyStack(scope constructs.Construct, id string, props *props.KMSKeyStackProps) (awscdk.Stack, error) {
+	var sprops awscdk.StackProps
+	if props != nil {
+		sprops = props.StackProps
+	}
+	stack := awscdk.NewStack(scope, &id, &sprops)
+
+	keySpec, keyUsage, err := props.GetKeySpecAndUsage()
+	if err != nil {
+		return nil, err
+	}
+
+	keyProps := &awskms.KeyProps{
+		Enabled:           jsii.Bool(props.Enabled),
+		EnableKeyRotation: jsii.Bool(props.EnableKeyRotation),
+		KeySpec:           keySpec,
+		KeyUsage:          keyUsage,
+		PendingWindow:     awscdk.Duration_Days(jsii.Number(props.PendingWindowDays)),
+
+		// Intentionally withhold the ability to configure the removal policy to avoid accidental key deletion.
```

Comment:
If a user goes out of their way to configure immediate removal could we really argue that it was accidental? I think protecting the user is a good default but we may not way to prevent them from shooting their foot if they want